### DISCUSSION TITLE
Email on Failure

### DISCUSSION THREAD

**Discussion by techwithanant** [jhuckaby/Cronicle#520] [Q&A] [answered]
Hi,
Thanks for creating Cronicle.
I have a quick question & would appreciate if someone can help me. I want to send email on any job failure. I also updated my config.json file after going through some issues and updated as below.

"base_app_url": "http://localhost:3012",
        "email_from": "xxxxxxxxxx@gmail.com",
        "smtp_hostname": "localhost",
        "smtp_port": 25,
        "secret_key": "xxxxxxxxxxxxx",
        "mail_options": {
                "secure": false,
                "auth": { "user": "xxxxxxxxxxxx@gmail.com", "pass": "xxxxxxxxxxxx" },
                "connectionTimeout": 10000,
                "greetingTimeout": 10000,
                "socketTimeout": 10000
        },

But I am still getting error - 
Failed to send e-mail for job: jl75pa7z307: xxxxxxxxxxxx@gmail.com: Error: connect ECONNREFUSED 127.0.0.1:25

**Comment by jhuckaby** [comment, marked as answer]
This is not an issue with Cronicle.  This is just server configuration and setup.  In order to send mail through localhost, like you have your configuration set, you need to run a local mail forwarding service such as postfix or sendmail, which needs to listen on port 25.

See: https://www.redhat.com/sysadmin/install-configure-postfix for RedHat / CentOS.

Or: https://www.digitalocean.com/community/tutorials/how-to-install-and-configure-postfix-on-ubuntu-20-04 for Ubuntu.

**Reply by techwithanant** [reply]
Thanks a lot Joseph for sharing this. I will looking into it.
